From 7339f767953ccd301e14f5a533c5629bfccdbebf Mon Sep 17 00:00:00 2001
From: Pete Bunting <petebunting@mac.com>
Date: Thu, 1 Dec 2022 08:28:30 +0000
Subject: [PATCH] Add gdal.lib to library list for windows. Name changed in
 GDAL 3.6.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fea45c8f..253b8871 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -293,7 +293,7 @@ endif(MSVC)
 
 include_directories(${GDAL_INCLUDE_DIR})
 if (MSVC)
-  set(GDAL_LIBRARIES -LIBPATH:${GDAL_LIB_PATH} gdal_i.lib)
+  set(GDAL_LIBRARIES -LIBPATH:${GDAL_LIB_PATH} gdal.lib)
 else()
   set(GDAL_LIBRARIES -L${GDAL_LIB_PATH} -lgdal)
 endif(MSVC)
-- 
2.37.1 (Apple Git-137.1)

